### PR TITLE
add .editorconfig config

### PR DESCRIPTION
**Pull request by alissonlauffer**
This pull request adds an .editorconfig file to the repository to standardize code formatting across most editors and IDEs.

Including this file will ensure a consistent formatting style regardless of the editor or IDE used. This will minimize unnecessary whitespace code diffs in pull requests (made by different editor configurations).

Also, having a final newline in files is a good practice for git, as adding new code at the end of the current code will produce a reduced diff.

### CODE REVIEW

**Comment by xou816** [comment]
Sounds reasonable :) 
Is there a way to apply the code style to the whole codebase automatically or will we have to go though every file at least once?

**Comment by alissonlauffer** [comment]
I'm afraid there isn't a tool for that, but I can do that job manually, as most files already follow this style guide (`cargo fmt` enforces trailing newlines and whitespace trimming by default).

File: src/main.rs [hunk 1]
```diff
@@ -112,9 +112,15 @@ fn setup_gtk(settings: &settings::SpotSettings) {
 }
 
 fn setup_credits(about: gtk::AboutDialog) {
-    let authors: Vec<&str> = include_str!("../AUTHORS").split('\n').collect();
-    let translators = include_str!("../TRANSLATORS");
-    let artists: Vec<&str> = include_str!("../ARTISTS").split('\n').collect();
+    let authors: Vec<&str> = include_str!("../AUTHORS")
+        .trim_end_matches('\n')
```
Comment (xou816) [inline src/main.rs:116]: :+1:

**Review by xou816 — APPROVED** [review]
looking good, thanks :)

**Pull request merged**